src/test/kotlin/nomic/unit/domain/auth/TokenRegistryTest.kt

```diff
@@ -0,0 +1,136 @@
+package nomic.unit.domain.auth
+
+import com.auth0.jwt.RegisteredClaims
+import nomic.data.repositories.IUserRepository
+import nomic.domain.auth.IKeyProvider
+import nomic.domain.auth.RSAKeyPair
+import nomic.domain.auth.TokenRegistry
+import nomic.domain.entities.User
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import java.security.KeyFactory
+import java.security.interfaces.RSAPrivateKey
+import java.security.interfaces.RSAPublicKey
+import java.security.spec.PKCS8EncodedKeySpec
+import java.security.spec.X509EncodedKeySpec
+import java.util.*
+
+class TokenRegistryTest {
+
+    private val keyProvider: IKeyProvider
+    private val usersRepo: IUserRepository
+
+    companion object {
+        const val PUBLIC_KEY = "MFwwDQYJKoZIhvcNAQEBBQADSwAwSAJBAI6jowrmGDTzQg7YQIyEoxQbvB16efH1EsEHj6yE58mIxXNi95ZEO5u7/loWmd1a1eWGI2dIu2dkDyYTVx8YDfUCAwEAAQ=="
+        const val PRIVATE_KEY = "MIIBVAIBADANBgkqhkiG9w0BAQEFAASCAT4wggE6AgEAAkEAjqOjCuYYNPNCDthAjISjFBu8HXp58fUSwQePrITnyYjFc2L3lkQ7m7v+WhaZ3VrV5YYjZ0i7Z2QPJhNXHxgN9QIDAQABAkAPPnXCSGz127mHlJzxZ4t2LrCgBeLadPKYJpK4QUv0QbzawRQxGJRmV2a+H6Y1oMOXvWhpZXQJIk2ff4EbInnRAiEAxVQdFxWP1QpE03wxeax1ZKHHKZyin+2ucdaue7mDThsCIQC5DMfjo8PX6hI/kl4F0/L+n6ur6Ngzy++OVOiQJwWVLwIgdr1is91Zq4x+VfRJoTnRejiPK88BXNMdQYs5e2HqTTkCIQCtKxGvrCPsXOZAyWblS0edEaykDhystA50gm6z32BxDwIgEaaS6Vy66nFFvX6X79YiZ5++aomx0Tl3d0+L0il2+GQ="
+
+        val publicKeyBytes = Base64.getDecoder().decode(PUBLIC_KEY)
+        val privateKeyBytes = Base64.getDecoder().decode(PRIVATE_KEY)
```

Comment:
Recommended by IDE to add explicit type since the current inference does not make it clear when working with the object